keyserver/listener.go

```diff
@@ -0,0 +1,158 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
+	defer server.waitStop.Done()
+	defer ln.Close()
+	go func() {
+		<-server.stop
+		ln.Close()
+	}()
+
+	for {
+		conn, err := ln.Accept()
+		if err != nil {
+			select {
+			case <-server.stop:
+				return
+			default:
+				log.Printf("accept client: %s", err)
+				continue
+			}
+		}
+		server.waitStop.Add(1)
+		go server.acceptClient(conn, handler)
+	}
+}
+
+func (server *ConiksServer) acceptClient(conn net.Conn, handler func(msg []byte) ([]byte, error)) {
+	defer conn.Close()
+	defer server.waitStop.Done()
+	closed := make(chan struct{})
+	defer close(closed)
+	go func() {
+		select {
+		case <-server.stop:
+			conn.Close()
+		case <-closed:
+		}
+	}()
+
+	conn.SetDeadline(time.Now().Add(5 * time.Second))
+
+	// handle request
+	buf := make([]byte, 4<<10)
```

Comment:
Maybe that's a dump question but why not simply write 4096 instead of `4<<10` for the buffer size?